File: python/pozdrav.py
def pozdrav(name, surname):
    print(f"Dobar dan, {surname} {name}")


def validiraj(text, polje):
    """Proverava da ime/prezime ima 3+ slova i nije broj."""
    if len(text) < 3:
        print(f"Greška: {polje} mora imati najmanje 3 slova.")
        return False
    if text.isdigit():
        print(f"Greška: {polje} ne sme biti broj.")
        return False
    return True


# unos od korisnika
while True:
    name = input("Unesi ime: ")
    surname = input("Unesi prezime: ")
    if validiraj(name, "Ime") and validiraj(surname, "Prezime"):
        break
    print("Pokušaj ponovo.\n")

pozdrav(name, surname)

File: typescript/pozdrav.ts
import * as readline from "readline";

function pozdrav(name: string, surname: string): void {
  console.log(`Dobar dan, ${surname} ${name}`);
}

function validiraj(text: string, polje: string): boolean {
  if (text.length < 3) {
    console.log(`Greška: ${polje} mora imati najmanje 3 slova.`);
    return false;
  }
  if (/^\d+$/.test(text)) {
    console.log(`Greška: ${polje} ne sme biti broj.`);
    return false;
  }
  return true;
}

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const ask = (question: string): Promise<string> =>
  new Promise((resolve) => rl.question(question, resolve));

(async () => {
  let name: string;
  let surname: string;
  while (true) {
    name = await ask("Unesi ime: ");
    surname = await ask("Unesi prezime: ");
    if (validiraj(name, "Ime") && validiraj(surname, "Prezime")) {
      break;
    }
    console.log("Pokušaj ponovo.\n");
  }
  pozdrav(name, surname);
  rl.close();
})();
